TombEngine/Shaders/Math.hlsli

```diff
@@ -29,4 +29,106 @@ float LinearizeDepth(float depth, float nearPlane, float farPlane)
 	return (2.0f * nearPlane) / (farPlane + nearPlane - depth * (farPlane - nearPlane));
 }
 
+float3 mod289(float3 x) {
```

Comment:
I see these 2 funcs are named "mod289" what does it mean? I am not sure if the shaders have to be readable :P.

Reply:
mod289 is function from The Book of Shaders, it is a permutation on the scale of 289 units, per 289 times